promql/parser/lex.go

```diff
@@ -28,6 +28,12 @@ type Item struct {
 	Val string   // The value of this Item.
 }
 
+func lexPrintln(a ...any) {
+	if yyDebug >= 3 {
```

Comment:
Despite this if statement, calling this function isn't free. In particular, the arguments are stringified before even reaching this. I'm not sure how performance critical the lexing is (some users have veeerrry long rules files), but conservatively, you should probably remove these calls before merging.